plugin/storage/badger/options.go

```diff
@@ -0,0 +1,124 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package badger
+
+import (
+	"flag"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// Options store storage plugin related configs
+type Options struct {
+	primary *NamespaceConfig
+	others  map[string]*NamespaceConfig
+}
+
+// NamespaceConfig is badger's internal configuration data
+type NamespaceConfig struct {
+	namespace      string
+	SpanStoreTTL   time.Duration
+	ValueDirectory string
+	KeyDirectory   string
+	Ephemeral      bool // Setting this to true will ignore ValueDirectory and KeyDirectory
+	SyncWrites     bool
+}
+
+const (
+	suffixKeyDirectory   = ".directory-key"
+	suffixValueDirectory = ".directory-value"
+	suffixEphemeral      = ".ephemeral"
+	suffixSpanstoreTTL   = ".span-store-ttl"
+	suffixSyncWrite      = ".consistency"
+)
+
+// NewOptions creates a new Options struct.
+func NewOptions(primaryNamespace string, otherNamespaces ...string) *Options {
+	options := &Options{
+		primary: &NamespaceConfig{
+			namespace:      primaryNamespace,
+			SpanStoreTTL:   time.Hour * 72, // Default is 3 days
+			SyncWrites:     false,          // Performance over consistency
+			Ephemeral:      true,           // Default is ephemeral storage
+			ValueDirectory: "",
+			KeyDirectory:   "",
+		},
+		others: make(map[string]*NamespaceConfig, len(otherNamespaces)),
+	}
+
+	for _, namespace := range otherNamespaces {
+		options.others[namespace] = &NamespaceConfig{namespace: namespace}
+	}
+
+	return options
+}
+
+// AddFlags adds flags for Options
+func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
+	addFlags(flagSet, opt.primary)
+	for _, cfg := range opt.others {
+		addFlags(flagSet, cfg)
+	}
+}
+
+func addFlags(flagSet *flag.FlagSet, nsConfig *NamespaceConfig) {
+	flagSet.Bool(
+		nsConfig.namespace+suffixEphemeral,
+		nsConfig.Ephemeral,
+		"Mark this storage ephemeral, data is stored in tmpfs (in-memory). Default is true.",
+	)
+	flagSet.Duration(
+		nsConfig.namespace+suffixSpanstoreTTL,
+		nsConfig.SpanStoreTTL,
+		"time.Duration to store the data. Default is 72 hours (3 days).",
+	)
+	flagSet.String(
+		nsConfig.namespace+suffixKeyDirectory,
+		nsConfig.KeyDirectory,
+		"Path to store the keys (indexes), this directory should reside in SSD disk. Set ephmeral to false if you want to define this setting.",
+	)
+	flagSet.String(
+		nsConfig.namespace+suffixValueDirectory,
+		nsConfig.ValueDirectory,
+		"Path to store the values (spans). Set ephmeral to false if you want to define this setting.",
+	)
+	flagSet.Bool(
+		nsConfig.namespace+suffixSyncWrite,
+		nsConfig.SyncWrites,
+		"If all writes should be synced immediately. This will greatly reduce write performance and will require fast SSD drives. Default is false.",
```

Comment:
> will require fast SSD drives

If I'm not on SSD, are you stopping the process? If not, what does "require" mean here?

Reply:
Mostly that this backend really wants SSDs and if you're going to sync all the writes, you probably want something with fast write time. I've removed it for now since technically you could use a DRAM backed disk array also for writes (but reads would be slow most likely)